### PR TITLE
Fix API to prevent users from uploading `<limits><eps>` section when disabled

### PR DESCRIPTION
**Pull request by Selutario**
|Related issue|
|---|
| #16419 |

## Description

This PR fixes the API `upload_configuration > limits > eps` configuration so it prevents users from uploading `ossec.conf` files containing `<limits><eps>` block inside [global](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/global.html#reference-ossec-global) tag when disabled in the `api.yaml`:
https://github.com/wazuh/wazuh/blob/607aaa80bb42416757c2f62335cc1cc8ff2095eb/api/api/configuration/api.yaml#L60-L71

Regex is now used to match `<limit><eps>` anywhere in `ossec.conf` uploaded by a user from the API, regardless of whether it is in a different `<ossec_config>` block and without breaking the current behavior of the endpoint.

#### EPS disallowed
```YAML
# Uploadable Wazuh configuration sections
 upload_configuration:
   limits:
     eps:
       allow: no
```
image
image

#### EPS allowed
```YAML
# Uploadable Wazuh configuration sections
 upload_configuration:
   limits:
     eps:
       allow: yes
```
image

## Tests
```
$ PYTHONPATH=/home/selu/Git/wazuh/api:/home/selu/Git/wazuh/framework python3 -m pytest framework/ -x --disable-warnings
============================================================================================ test session starts ============================================================================================
platform linux -- Python 3.9.16, pytest-7.0.1, pluggy-0.13.1
rootdir: /home/selu/Git/wazuh/framework
plugins: anyio-3.6.2, asyncio-0.18.1, aiohttp-1.0.4, cov-4.0.0
asyncio: mode=auto
collected 2047 items                                                                                                                                                                                        

framework/scripts/tests/test_agent_groups.py ..............                                                                                                                                           [  0%]
framework/scripts/tests/test_agent_upgrade.py ...............                                                                                                                                         [  1%]
framework/scripts/tests/test_cluster_control.py ......                                                                                                                                                [  1%]
framework/scripts/tests/test_wazuh_clusterd.py .......                                                                                                                                                [  2%]
framework/scripts/tests/test_wazuh_logtest.py ......................                                                                                                                                  [  3%]
framework/wazuh/core/cluster/dapi/tests/test_dapi.py ................................                                                                                                                 [  4%]
framework/wazuh/core/cluster/tests/test_client.py ................                                                                                                                                    [  5%]
framework/wazuh/core/cluster/tests/test_cluster.py ...................................                                                                                                                [  7%]
framework/wazuh/core/cluster/tests/test_common.py ...........................................................................                                                                         [ 10%]
framework/wazuh/core/cluster/tests/test_control.py ......                                                                                                                                             [ 11%]
framework/wazuh/core/cluster/tests/test_local_client.py .............                                                                                                                                 [ 11%]
framework/wazuh/core/cluster/tests/test_local_server.py ........................                                                                                                                      [ 12%]
framework/wazuh/core/cluster/tests/test_master.py .............................................                                                                                                       [ 15%]
framework/wazuh/core/cluster/tests/test_server.py ...................                                                                                                                                 [ 16%]
framework/wazuh/core/cluster/tests/test_utils.py ...........                                                                                                                                          [ 16%]
framework/wazuh/core/cluster/tests/test_worker.py ....................................                                                                                                                [ 18%]
framework/wazuh/core/tests/test_active_response.py ..............                                                                                                                                     [ 19%]
framework/wazuh/core/tests/test_agent.py ................................................................................................................................................             [ 26%]
framework/wazuh/core/tests/test_cdb_list.py ......................................                                                                                                                    [ 27%]
framework/wazuh/core/tests/test_common.py .......                                                                                                                                                     [ 28%]
framework/wazuh/core/tests/test_configuration.py ......................................................................                                                                               [ 31%]
framework/wazuh/core/tests/test_database.py .............                                                                                                                                             [ 32%]
framework/wazuh/core/tests/test_decoder.py ................                                                                                                                                           [ 33%]
framework/wazuh/core/tests/test_exception.py ...                                                                                                                                                      [ 33%]
framework/wazuh/core/tests/test_input_validator.py ...                                                                                                                                                [ 33%]
framework/wazuh/core/tests/test_logtest.py ..                                                                                                                                                         [ 33%]
framework/wazuh/core/tests/test_manager.py ...............                                                                                                                                            [ 34%]
framework/wazuh/core/tests/test_mitre.py .............                                                                                                                                                [ 34%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                                                                                                                                               [ 35%]
framework/wazuh/core/tests/test_results.py ........................................                                                                                                                   [ 37%]
framework/wazuh/core/tests/test_rootcheck.py .............                                                                                                                                            [ 37%]
framework/wazuh/core/tests/test_rule.py ......................                                                                                                                                        [ 38%]
framework/wazuh/core/tests/test_sca.py ........................                                                                                                                                       [ 39%]
framework/wazuh/core/tests/test_security.py ............                                                                                                                                              [ 40%]
framework/wazuh/core/tests/test_stats.py .................                                                                                                                                            [ 41%]
framework/wazuh/core/tests/test_syscheck.py .......                                                                                                                                                   [ 41%]
framework/wazuh/core/tests/test_syscollector.py ...                                                                                                                                                   [ 41%]
framework/wazuh/core/tests/test_task.py ........                                                                                                                                                      [ 42%]
framework/wazuh/core/tests/test_utils.py ............................................................................................................................................................ [ 49%]
.....................................................................................                                                                                                                 [ 54%]
framework/wazuh/core/tests/test_vulnerability.py ..                                                                                                                                                   [ 54%]
framework/wazuh/core/tests/test_wazuh_queue.py ....................                                                                                                                                   [ 55%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................                                                                                                                                  [ 56%]
framework/wazuh/core/tests/test_wdb.py ...............................                                                                                                                                [ 57%]
framework/wazuh/core/tests/test_wlogging.py .........                                                                                                                                                 [ 58%]
framework/wazuh/rbac/tests/test_auth_context.py ..                                                                                                                                                    [ 58%]
framework/wazuh/rbac/tests/test_decorators.py .............................................................................................................                                           [ 63%]
framework/wazuh/rbac/tests/test_default_configuration.py ........................................................                                                                                     [ 66%]
framework/wazuh/rbac/tests/test_orm.py ......................................................                                                                                                         [ 68%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........                                                                                                                                           [ 69%]
framework/wazuh/tests/test_active_response.py ............                                                                                                                                            [ 69%]
framework/wazuh/tests/test_agent.py .....................................................................................................................                                             [ 75%]
framework/wazuh/tests/test_cdb_list.py .....................................................                                                                                                          [ 78%]
framework/wazuh/tests/test_ciscat.py .................................                                                                                                                                [ 79%]
framework/wazuh/tests/test_cluster.py ..........                                                                                                                                                      [ 80%]
framework/wazuh/tests/test_decoder.py ...................................                                                                                                                             [ 82%]
framework/wazuh/tests/test_group.py .......                                                                                                                                                           [ 82%]
framework/wazuh/tests/test_logtest.py ......                                                                                                                                                          [ 82%]
framework/wazuh/tests/test_manager.py ....................................                                                                                                                            [ 84%]
framework/wazuh/tests/test_mitre.py .......                                                                                                                                                           [ 84%]
framework/wazuh/tests/test_rootcheck.py ..................................................                                                                                                            [ 87%]
framework/wazuh/tests/test_rule.py ........................................................                                                                                                           [ 89%]
framework/wazuh/tests/test_sca.py ...........                                                                                                                                                         [ 90%]
framework/wazuh/tests/test_security.py .........................................................................                                                                                      [ 94%]
framework/wazuh/tests/test_stats.py ...............                                                                                                                                                   [ 94%]
framework/wazuh/tests/test_syscheck.py ..........................                                                                                                                                     [ 96%]
framework/wazuh/tests/test_syscollector.py ............                                                                                                                                               [ 96%]
framework/wazuh/tests/test_task.py ............................                                                                                                                                       [ 98%]
framework/wazuh/tests/test_vulnerability.py ........................................                                                                                                                  [100%]

=============================================================================== 2047 passed, 43 warnings in 342.60s (0:05:42) ===============================================================================
```

### CODE REVIEW

File: framework/wazuh/core/utils.py [hunk 1]
```diff
@@ -791,18 +791,14 @@ def check_disabled_limits_in_conf(data):
...
-    xml_file = fromstring(f"<root_tag>{data}</root_tag>")
-    found_limits = []
-    for global_section in xml_file.findall("global"):
-        found_limits += [limit_section for limit_section in global_section.findall("limits") or []]
-    if len(found_limits) == 0:
-        return
+    def check_section(section_name):
+        if re.findall(pattern=r'<global>.*<limits>.*<{0}>.*</{0}>.*</limits>.*</global>'.format(section_name),
+                      string=data, flags=re.MULTILINE | re.DOTALL | re.IGNORECASE):
+            raise WazuhError(1127, extra_message=f"global > limits > {disabled_limit}")
```
Comment (nico-stefani) [inline framework/wazuh/core/utils.py:797]: I think it should be `{section_name}` instead of `{disabled_limit}`.

**Review by nico-stefani — CHANGES_REQUESTED** [review]

File: framework/wazuh/core/utils.py (hunk 1 above)
Comment (Selutario) [reply framework/wazuh/core/utils.py]: Fixed in https://github.com/wazuh/wazuh/pull/16455/commits/962ce760e1e3ccff949b50ebac8f1393abf09d67

**Review by nico-stefani — APPROVED** [review]
LGTM

**Review by davidjiglesias — APPROVED** [review]

**Review by jmv74211 — APPROVED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/4049
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/4049#issuecomment-1482579036

**Pull request merged**